MyBus/MyBus/Model/Road/MapBusRoad.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  MapBusRoad.swift
+//  MyBus
+//
+//  Created by Lisandro Falconi on 5/9/16.
+//  Copyright © 2016 Spark Digital. All rights reserved.
+//
+
+import Foundation
+import Mapbox
+
+class MapBusRoad: NSObject {
+    var markerList : [MGLPointAnnotation] = [MGLPointAnnotation]()
+    var polyLineList : [MGLPolyline] = [MGLPolyline]()
+
+    func addBusRoadOnMap(roadResult : RoadResult) -> MapBusRoad
+    {
+        if(roadResult.roadResultType == 0)
+        {
+            // Declare the marker point and set its coordinates
+            let mapPoint = MGLPointAnnotation()
```

Comment:
Another refactor to consider here is extract if and else blocks and move them to new methods.